qiskit_optimization/runtime/qaoa_program.py

```diff
@@ -0,0 +1,159 @@
+        provider: Optional[Provider] = None,
+        backend: Optional[Backend] = None,
+        shots: int = 1024,
+        measurement_error_mitigation: bool = False,
+        callback: Optional[Callable[[int, np.ndarray, float, float], None]] = None,
+        store_intermediate: bool = False,
+    ) -> None:
+        """
+        Args:
+            optimizer: A dictionary specifying a classical optimizer.
+                Currently only SPSA and QN-SPSA are supported. Per default, SPSA is used.
+                The dictionary must contain a key ``name`` for the name of the optimizer and may
+                contain additional keys for the settings.
+                E.g. ``{'name': 'SPSA', 'maxiter': 100}``.
+            reps: the integer parameter :math:`p` as specified in https://arxiv.org/abs/1411.4028,
+                Has a minimum valid value of 1.
+            initial_state: An optional initial state to prepend the QAOA circuit with
+            mixer: the mixer Hamiltonian to evolve with or a custom quantum circuit. Allows support
+                of optimizations in constrained subspaces as per https://arxiv.org/abs/1709.03489
+                as well as warm-starting the optimization as introduced
+                in http://arxiv.org/abs/2009.10095.
+            initial_point: An optional initial point (i.e. initial parameter values)
+                for the optimizer. If ``None`` a random vector is used.
+            provider: The provider.
+            backend: The backend to run the circuits on.
+            shots: The number of shots to be used
+            measurement_error_mitigation: Whether or not to use measurement error mitigation.
+            callback: a callback that can access the intermediate data during the optimization.
+                Four parameter values are passed to the callback as follows during each evaluation
+                by the optimizer for its current set of parameters as it works towards the minimum.
+                These are: the evaluation count, the optimizer parameters for the
+                ansatz, the evaluated mean and the evaluated standard deviation.
+            store_intermediate: Whether or not to store intermediate values of the optimization
+                steps. Per default False.
+        """
+        super().__init__(
+            ansatz=None,
+            optimizer=optimizer,
+            initial_point=initial_point,
+            provider=provider,
+            backend=backend,
+            shots=shots,
+            measurement_error_mitigation=measurement_error_mitigation,
+            callback=callback,
+            store_intermediate=store_intermediate,
+        )
+        self._initial_state = initial_state
+        self._mixer = mixer
+        self._reps = reps
+
+    @property
+    def ansatz(self) -> Optional[QuantumCircuit]:
+        return self._ansatz
+
+    @ansatz.setter
+    def ansatz(self, ansatz: QuantumCircuit) -> None:
+        raise NotImplementedError(
```

Comment:
Is this the right exception to raise in this case?

Reply:
i replaced it with a `QiskitOptimizationError` since I didn't find any other satisfying replacement 🙂